### PR TITLE
8326705: Test CertMsgCheck.java fails to find alert certificate_required 

### PR DESCRIPTION
**Pull request by ascarpino**
Hi,

I need a review for this simple change to fix a threading problem with the test. The server thread was not completing before the check occurred on the main thread.  The failure showed up in windows and macos, but not linux.  With this fix, running 100 times, windows & macos showed no failures.

Tony

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326705](https://bugs.openjdk.org/browse/JDK-8326705): Test CertMsgCheck.java fails to find alert certificate_required (**Bug** - P3)

### Reviewers
 * [Sibabrata Sahoo](https://openjdk.org/census#ssahoo) (@sisahoo - Committer)
 * [Rajan Halade](https://openjdk.org/census#rhalade) (@rhalade - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19553/head:pull/19553` \
`$ git checkout pull/19553`

Update a local copy of the PR: \
`$ git checkout pull/19553` \
`$ git pull https://git.openjdk.org/jdk.git pull/19553/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19553`

View PR using the GUI difftool: \
`$ git pr show -t 19553`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19553.diff">https://git.openjdk.org/jdk/pull/19553.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19553#issuecomment-2150468317)

### CODE REVIEW

File: test/jdk/javax/net/ssl/templates/TLSBase.java [hunk 1]
```diff
@@ -97,8 +97,8 @@ public void write(SSLSocket sock, byte[] data) throws Exception {
     private static KeyManager[] getKeyManager(boolean empty) throws Exception {
         FileInputStream fis = null;
         if (!empty) {
-            fis = new FileInputStream(System.getProperty("test.src", "./") + "/" + pathToStores +
-                "/" + keyStoreFile);
+            fis = new FileInputStream(System.getProperty("test.src", "./") +
+                "/" + pathToStores + "/" + keyStoreFile);
```
Comment (mcpowers) [inline test/jdk/javax/net/ssl/templates/TLSBase.java:101]: My Java style guide says to break before the operator.

File: test/jdk/javax/net/ssl/templates/TLSBase.java [hunk 2]
```diff
@@ -239,7 +243,7 @@ static class Server extends TLSBase {
         // test or the test will never end.
         void done() {
             try {
-                t.interrupt();
+                t.join(5000);
```
Comment (mcpowers) [inline test/jdk/javax/net/ssl/templates/TLSBase.java:246]: I had to increase the join value of a test recently that failed with virtual threads.
Try adding `--test-make-args JTREG_TEST_THREAD_FACTORY=Virtual` to your mach5 command line.

File: test/jdk/javax/net/ssl/SSLSession/CertMsgCheck.java [hunk 3]
```diff
@@ -40,20 +40,25 @@ public static void main(String[] args) throws Exception {
...
-                }
+                return;
+            } else {
+                System.out.println("No \"" + args[0] + "\" found.");
             }
-
-            throw new Exception("Failed to find expected alert: " + args[0]);
         }
+
+        throw new Exception("Failed to find expected alert: " + args[0]);
```
[the remark this answers is not in the document]
Comment (ascarpino) [reply test/jdk/javax/net/ssl/SSLSession/CertMsgCheck.java:62]: It's the TLS alert it's trying to find, the exception is the messenger

File: test/jdk/javax/net/ssl/templates/TLSBase.java (hunk 1 above)
Comment (ascarpino) [reply test/jdk/javax/net/ssl/templates/TLSBase.java:101]: I have never heard that.  I have always ended with a plus and I believe everyone else does too if I'm not mistaken.

File: test/jdk/javax/net/ssl/templates/TLSBase.java (hunk 2 above)
Comment (ascarpino) [reply test/jdk/javax/net/ssl/templates/TLSBase.java:246]: I'll check it out.. thanks

File: test/jdk/javax/net/ssl/templates/TLSBase.java (hunk 1 above)
Comment (mcpowers) [reply test/jdk/javax/net/ssl/templates/TLSBase.java:101]: https://cr.openjdk.org/~alundblad/styleguide/index-v6.html#toc-constants

Probably showed up in a code review comment from someone.

File: test/jdk/javax/net/ssl/templates/TLSBase.java (hunk 2 above)
Comment (ascarpino) [reply test/jdk/javax/net/ssl/templates/TLSBase.java:246]: Test passed with virtual threads on linux, windows, & mac aarch64

File: test/jdk/javax/net/ssl/templates/TLSBase.java (hunk 1 above)
Comment (ascarpino) [reply test/jdk/javax/net/ssl/templates/TLSBase.java:101]: There are a lot of places where that style is violated.  Looking at just a few random files I didn't see any following the correct style (no scientific study).  I'm going to leave it as is.

**Review by sisahoo — APPROVED** [review]

**Review by rhalade — APPROVED** [review]

**Pull request closed without merging**